src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/GetAllTest.java

```diff
@@ -34,35 +39,62 @@
-            .andExpect(content().json(emptyResponse));
-        sendGet("/jobs?size=1")
-            .andExpect(status().isOk())
-            .andExpect(content().json(emptyResponse));
-        sendGet("/jobs?page=1&size=1")
-            .andExpect(status().isOk())
-            .andExpect(content().json(emptyResponse));
+        Arrays.asList(
+            "/jobs",
+            "/jobs?page=1",
+            "/jobs?size",
+            "/jobs?page=1&size=1"
+        ).forEach(url -> {
+            try {
+                sendGet(url)
+                    .andExpect(status().isOk())
+                    .andExpect(jsonPath("content").isArray())
+                    .andExpect(jsonPath("content").isEmpty())
+                    .andExpect(jsonPath("last").value(true))
+                    .andExpect(jsonPath("first").value(true));
+            } catch (Exception exc) {
+                fail("Failed to test '" + url + "'", exc);
+            }
+        });
     }
 
     @Test
-    public void should_return_some_data_when_non_empty_list_is_returned() throws Exception {
+    public void should_return_default_pages_when_no_parameters_are_passed() throws Exception {
         JobData jobData = JobData.fromJob("some-id", validJob());
-        when(jobsService.getAll(anyString())).thenReturn(new JobList(Collections.singletonList(jobData)));
+        when(jobsService.getAll(anyString(), anyInt(), anyInt()))
+            .thenReturn(new Pages<>(Collections.singletonList(jobData)));
 
         sendGet()
             .andExpect(status().isOk())
-            .andExpect(jsonPath("data[0].action").exists())
-            .andExpect(jsonPath("data[0].id").value(jobData.id))
-            .andExpect(jsonPath("data[0].name").value(jobData.name));
+            .andExpect(jsonPath("content[0].action").exists())
+            .andExpect(jsonPath("content[0].id").value(jobData.id))
+            .andExpect(jsonPath("content[0].name").value(jobData.name))
+            .andExpect(jsonPath("total_pages").value(1))
+            .andExpect(jsonPath("total_elements").value(1))
+            .andExpect(jsonPath("number_of_elements").value(1));
+    }
+
+    @Test
+    public void should_paginate_response_content_correctly_according_to_parameters() throws Exception {
+        // given we have 3 jobs in total
+
+        getResponse(1, 10)
+            .andExpect(jsonPath("content").isEmpty())
+            .andExpect(jsonPath("total_pages").value(1))
+            .andExpect(jsonPath("number_of_elements").value(0));
+        getResponse(0, 2)
+            .andExpect(jsonPath("total_pages").value(2))
+            .andExpect(jsonPath("number_of_elements").value(2));
+        getResponse(1, 2)
+            .andExpect(jsonPath("total_pages").value(2))
+            .andExpect(jsonPath("number_of_elements").value(1));
+        getResponse(1, 1)
+            .andExpect(jsonPath("total_pages").value(3))
+            .andExpect(jsonPath("number_of_elements").value(1));
     }
```

Comment:
I think we should also test edge cases (like negative values, page index beyond number of pages available etc)

Reply:
beyond availability is there. i'll add the negative